Add C# customer-profile rest service (#188)
vmware-tanzu/application-accelerator-samples: csharp-rest-service/README.md

# Introduction

> csharp-rest-service provides you an out-of-the-box example of a .NET WebAPI application with PostgreSQL backend.

Reviewer: ```suggestion
csharp-rest-service provides you an out-of-the-box example of a .NET WebAPI application with a PostgreSQL backend.

```

The application also includes several Steeltoe management endpoints (including a database health check) and dynamic log level management.

For more details on Steeltoe, visit <https://docs.steeltoe.io/>.

## Running the app locally

To run the sample application:

```script
dotnet run
```

## Database

You will need a local database running, see [DATABASE.md](DATABASE.md#local).

## Deploying to Kubernetes as a TAP workload with Tanzu CLI

If you make modifications to the source, push these changes to your own Git repository.

> When you are done developing your app, you can simply deploy it using:

Reviewer: ```suggestion
Deploy your app from your Git repository by running:
```

```script
tanzu apps workload apply -f config/workload.yaml
```

> If you would like deploy the code from your local working directory you can use the following command:

Reviewer: ```suggestion
Deploy your app from your local working directory by running:
```

```script
tanzu apps workload create csharp-rest-service -f config/workload.yaml --local-path . --source-image <REPOSITORY-PREFIX>/csharp-rest-service-source --type web
```

## Accessing the app deployed to your cluster

Determine the URL to use for the accessing the app by running:

```script
tanzu apps workload get csharp-rest-service
```

> To access the deployed app use the URL shown under "Workload Knative Services" and append the endpoint `/api/customer-profiles` to that URL.

Reviewer: ```suggestion
To access the deployed app, use the URL shown under "Workload Knative Services", appending the endpoint `/api/customer-profiles`.
```
